RFCs/FS-1063-support-letbang-andbang-for-applicative-functors.md

```diff
@@ -0,0 +1,432 @@
+                                )
+                            )
+                        )),
+                    foo),
+                bar),
+            baz),
+        ce.Apply(
+            ce.Apply(
+                ce.Return(
+                    (fun x ->
+                        (fun y ->
+                            x + y
+                        )
+                    )),
+                foo),
+            bar)
+    ),
+    ce.Apply(
+        ce.Apply(
+            ce.Return(
+                (fun y ->
+                    (fun z ->
+                        y + z
+                    )
+                )),
+            bar),
+        baz)
+)
+```
+
+*N.B.* this syntax forces the writer to be explicit about how many times `Apply` should be called, and with which arguments, for each call to `Combine`. Notice also how the right-hand sides are still repeated for each alternative case in order to keep the occurrence of potential side-effects from evaluating them predictable, and also occur before the pattern matching _each time_ a new alternative case is explored.
+
+If this syntax feels a bit heavy, remember that the `yield` keyword is not required, and the new syntax can be mixed with other styles (e.g. a custom `<|>` alternation operator) to strike an appropriate balance as each situation requires.
+
+## Using
+
+Just as monads support `Using` via `use!`, applicatives supports it via `use! ... anduse! ...`. Each binding can be either `and!` or `anduse!` (unless it is the first, in which case it must be either `let!` or `use!`), i.e. you can mix-and-match to describe which bindings should be covered by a call to `Using`:
+
+```fsharp
+ce {
+     use! x    = foo // x is wrapped in a call to ce.Using(...)
+     and! y    = bar // y is _not_ wrapped in a call to ce.Using(...)
+     anduse! z = baz // z is wrapped in a call to ce.Using(...)
+     return x + y + z
+ }
+```
+
+?
+
+```fsharp
+ce.Apply(
+    ce.Apply(
+        ce.Apply(
+            ce.Return(
+                (fun x ->
+                    ce.Using(x, fun x ->
+                        (fun y ->                    // <- N.B. No ce.Using(...) call here because we used `and!`
+                            (fun z ->                // instead of `anduse!` for `y` in the CE. Similarly, we
+                                ce.Using(z, fun z -> // could have chose to use `let!` instead of `use!` for the
+                                    x + y + z        // first binding to avoid a call to Using
+                                )
+                            )
+                        )
+                    )
+                )),
+            foo), 
+        bar),
+    baz)
+```
+
+## Ambiguities surrounding a `let! .. return ...`
+[singlelet]: #singlelet
+
+Some CEs could be validly desugared in multiple ways, depending on which methods are implemented on a builder (assuming the implementations follow the standard laws relating these functions).
```

Comment:
Please omit "validly" from this:

"Some CEs could be validly desugared in multiple ways[...]"

It reads cleaner and basically says the same thing.